g4simple.cc

```diff
@@ -581,6 +589,45 @@ class G4SimpleRunManager : public G4RunManager, public G4UImessenger
           if(!doMatching || regex_match(name, pattern)) cout << name << ' ' << iRep << endl;
         }
       }
+
+      else if (command == fSetStepLimitCmd) {
+        istringstream iss(newValues);
+        G4String valueWithUnit, volNameRegex;
+        // Read the value and unit as separate strings, then combine them
+        G4double value;
+        G4String unit;
+        iss >> value >> unit;
+        valueWithUnit = G4String(std::to_string(value)) + unit;
+        // Check if there's argument for the volume name regex
+        std::getline(iss >> std::ws, volNameRegex);
+        // Apply the step limit using the parsed values
+        ApplyStepLimit(valueWithUnit, volNameRegex);
+      }
+    }
+
+    void ApplyStepLimit(const G4String& lengthAndUnit, const G4String& volNameRegex) {
+      // Split the string into value and unit
+      istringstream iss(lengthAndUnit);
+      G4double value;
+      G4String unit;
+      iss >> value >> unit;
+      // Convert the unit to a multiplier
+      G4double unitMultiplier = G4UnitDefinition::GetValueOf(unit);
+      // Convert the input value to mm using the unit multiplier
+      G4double g4_step_max = value * unitMultiplier;
```

Comment:
Move this G4String parsing up into SetNewValue() and make the signature 
`void ApplyStepLimit(G4double g4_step_max, const G4String& volNameRegex)`

Reply:
Thanks I made the changes